playFeature/src/main/java/com/panhuk/playfeature/PlayViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package com.panhuk.playfeature
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.panhuk.core.ERROR
+import com.panhuk.domain.model.Question
+import com.panhuk.repository.SessionTokenRepoReader
+import com.panhuk.useCase.GetQuestionsUseCase
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import java.util.Queue
+import javax.inject.Inject
+
+class PlayViewModel @Inject constructor(
+  private val getQuestionsUseCase: GetQuestionsUseCase,
+  private val sessionTokenRepoReader: SessionTokenRepoReader,
+  private val dispatcher: CoroutineDispatcher
+) : ViewModel() {
+
+  private lateinit var questions: Queue<Question>
+  private lateinit var sessionToken: String
+
+  var title: String = ""
+  var questionAnswers: List<String> = listOf()
+  var totalScore: Int = 0
+
+  init {
+    viewModelScope.launch(dispatcher) {
+      try {
+        generateNewSessionToken()
+        getQuestions()
+        loadQuestion()
+      } catch (e: Exception) {
+        Log.e(ERROR, e.toString())
```

Comment:
I meant error dialogs / Toasts or something that users can interact with. But that's fine/ Let's implement dialogs in further PR's